metabase/metabase
# Show new syncing tables as greyed out in data reference page

Closes #33152

### Description

Disables and grays out links to not-fully-synced tables in data reference page (`/reference`).

Design doc (this was not in the design doc though - most likely accidentally): https://www.notion.so/metabase/Provide-progress-on-initial-database-sync-16af8e79312a45ecb715d8e58801c52b#5d67f984521843c4818e17a37719a111

This is similar to #33076.

### How to verify

1. Simulate long sync with this patch (as shown in [this video](https://www.loom.com/share/e3e9b5a2c2a441aaba4c94ca4f01b776) from [this PR](https://github.com/metabase/metabase/pull/31609)) - _make sure to rebuild_:
   ```diff
   diff --git a/src/metabase/sync/sync_metadata/fields.clj b/src/metabase/sync/sync_metadata/fields.clj
   index 58147f61eb..55afecc5b0 100644
   --- a/src/metabase/sync/sync_metadata/fields.clj
   +++ b/src/metabase/sync/sync_metadata/fields.clj
   @@ -80,6 +80,7 @@
                                    :total-fields   su/IntGreaterThanOrEqualToZero})
      "Sync the Fields in the Metabase application database for all the Tables in a `database`."
      [database :- i/DatabaseInstance]
   +  (Thread/sleep 30000)
      (->> database
         sync-util/db->sync-tables
         (map (partial sync-fields-for-table! database))
   ```
2. Spin up the QA dbs:
    ```Shell
    yarn test-qa-dbs:up
    ````
3. Go to http://localhost:3000/admin/databases/create and connect Postgres QA DB ([credentials](https://github.com/metabase/metabase/blob/d6473cdce794254293e4be2cc4f17aef527ccf05/e2e/support/cypress_data.js#L164-L170)).
4. Now, within 30 seconds, open http://localhost:3000/reference and navigate to see the tables of the database connected in previous step

Expected result: the tables are grayed out and non-interactive. There is no hover effect on titles of disabled tables.

### Demo

#### Before

![image](https://github.com/metabase/metabase/assets/6830683/128305be-2844-420c-82f3-f9fccad5efad)

#### After

![image](https://github.com/metabase/metabase/assets/6830683/6e418088-c440-4c53-b9b4-eabf89291083)
